ui/src/components/Header.tsx

```diff
@@ -0,0 +1,20 @@
+import { t } from 'i18next';
+import { useAuthState } from 'react-firebase-hooks/auth';
+import { auth } from '@lib/firebase';
+import { useLocation } from 'react-router-dom';
+
+const Header = () => {
+  const [user] = useAuthState(auth);
+  const username = user?.email?.split('@')[0];
+  const location = useLocation();
+
+  return (
+    <div className="mb-4 flex flex-col leading-none">
+      {t('header.welcome')}
+      <span className="block text-xl font-bold">{username}</span>
+      {t('header.mainPage')} {location.key}
```

Comment:
Key za każdym rebuildem się zmienia. Jedynie co można zrobić to pathname albo state dla każdego route.

Reply:
no to możemy użyć pathname. Mamy już zdefiniowany obiekt z wszystkimi ścieżkami to możemy na jego podstawie stworzyc kolejny obiekt który będzie wskazywał na klucze translacji. Coś typu 

```ts
import paths from '@routes/paths';

export const pageNames = {
 [paths.PLAYGROUND]: 'pageNames.palyground',
  [paths.TASKSTODO]: 'pageNames.taskTodo',
} as const
```

i potem użyć go tutaj 

```suggestion
      {t(pageNames[location.path])}
```